subprojects/groovy-parser-antlr4/src/main/java/org/apache/groovy/parser/antlr4/TryWithResourcesASTTransformation.java

```diff
@@ -342,7 +348,7 @@ private ExpressionStatement createAddSuppressedStatement(String primaryExcName,
                 new MethodCallExpression(
                         new VariableExpression(primaryExcName),
                         "addSuppressed",
-                        new ArgumentListExpression(Collections.singletonList(new VariableExpression(suppressedExcName))));
+                        new ArgumentListExpression(Collections.singletonList((Expression) new VariableExpression(suppressedExcName))));
```

Comment:
Gotcha.
P.S.  I deleted the comment "unnecessary casting type" by accident... it's hard to edit comments on my cell phone.